src/internal/m365/controller_test.go

```diff
@@ -283,7 +283,7 @@ func (suite *ControllerIntegrationSuite) SetupSuite() {
 	ctx, flush := tester.NewContext(t)
 	defer flush()
 
-	suite.ctrl = newController(ctx, t, resource.Users)
+	suite.ctrl = newController(ctx, t, resource.Users, path.ExchangeService)
```

Comment:
Not a blocker, but we should add integration tests with OD & SP services too

Reply:
I don't think that's needed, imo.  It'd be testing a single conditional, and we'd have to jump through hoops to extract signal on the different results.